flexbox/src/main/java/com/google/android/flexbox/FlexboxLayout.java

```diff
@@ -715,59 +751,67 @@ private int shrinkFlexItems(FlexLine flexLine, @FlexDirection int flexDirection,
             LayoutParams lp = (LayoutParams) child.getLayoutParams();
             if (flexDirection == FLEX_DIRECTION_ROW
                     || flexDirection == FLEX_DIRECTION_ROW_REVERSE) {
-                float rawCalculatedWidth = child.getMeasuredWidth() - unitShrink * lp.flexShrink;
-                if (i == flexLine.itemCount - 1) {
-                    rawCalculatedWidth += accumulatedRoundError;
-                    accumulatedRoundError = 0;
-                }
-                int newWidth = Math.round(rawCalculatedWidth);
-                if (newWidth < 0) {
-                    // This means the child doesn't have enough space to distribute the negative
-                    // free space. To adjust the flex line length down to the maxMainSize, remaining
-                    // negative free space needs to be re-distributed to other flex items
-                    // (children views). In that case, invoke this method again with the same
-                    // startIndex.
-                    needsReshrink = true;
-                    newWidth = 0;
-                } else {
-                    accumulatedRoundError += (rawCalculatedWidth - newWidth);
-                    if (accumulatedRoundError > 1.0) {
-                        newWidth += 1;
-                        accumulatedRoundError -= 1;
-                    } else if (accumulatedRoundError < -1.0) {
-                        newWidth -= 1;
-                        accumulatedRoundError += 1;
+                if (!mFrozenChildren[childIndex]) {
```

Comment:
Regardless of if the child is frozen or not it still needs to run
```java
flexLine.mainSize += child.getMeasuredWidth() + lp.leftMargin + lp.rightMargin;
```
or
```java
flexLine.mainSize += child.getMeasuredHeight() + lp.topMargin + lp.bottomMargin;
```
for the main size to be correct. I'm all for reducing intention but I don't see a clean way to do this without duplicating those lines.